File: branches/Medusa/Source/Fantasy.Jobs/IJob.cs
using System;
using System.Xml.Linq;
namespace Fantasy.Jobs
{
    public interface IJob
    {
        TaskItem AddTaskItem(string name, string category);
        void RemoveTaskItem(TaskItem item);

        TaskItemGroup AddTaskItemGroup();
        void RemoveTaskItemGroup(TaskItemGroup group);


        TaskItem[] GetEvaluatedItemsByCatetory(string category);

        TaskItem GetEvaluatedItemByName(string name);

        TaskItem[] Items { get; }

        Guid ID { get; }

        string TemplateName { get;}

        JobProperty[] Properties { get; }
        string GetProperty(string name);
        
        void SetProperty(string name, string value);

        bool HasProperty(string name);
        
        void RemoveProperty(string name);
        string StartupTarget { get; set; }

        Type ResolveInstructionType(XName name);

        RuntimeStatus RuntimeStatus { get; }

        void Execute();

        void ExecuteInstruction(IInstruction instruction);

        void ExecuteTarget(string targetName);

        

    }
}

File: branches/Stheno/AutoConverted/Fantasy.Jobs/IJob.java
package Fantasy.Jobs;

public interface IJob
{
	TaskItem AddTaskItem(String name, String category);
	void RemoveTaskItem(TaskItem item);

	TaskItemGroup AddTaskItemGroup();
	void RemoveTaskItemGroup(TaskItemGroup group);


	TaskItem[] GetEvaluatedItemsByCatetory(String category);

	TaskItem GetEvaluatedItemByName(String name);

	TaskItem[] getItems();

	Guid getID();

	String getTemplateName();

	JobProperty[] getProperties();
	String GetProperty(String name);

	void SetProperty(String name, String value);

	boolean HasProperty(String name);

	void RemoveProperty(String name);
	String getStartupTarget();
	void setStartupTarget(String value);

	java.lang.Class ResolveInstructionType(XName name);

	RuntimeStatus getRuntimeStatus();

	void Execute();

	void ExecuteInstruction(IInstruction instruction);

	void ExecuteTarget(String targetName);



}